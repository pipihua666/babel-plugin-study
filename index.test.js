// index.test.js
const pluginTester = require('babel-plugin-tester').default
const myPlugin = require('./index')

pluginTester({
  plugin: myPlugin,
  tests: [
    {
      code: 'a**b',
      output: 'Math.pow(a, b);',
      title: '操作符'
    },
    {
      code: 'a**=b',
      snapshot: true,
      title: '赋值'
    }
  ]
})
