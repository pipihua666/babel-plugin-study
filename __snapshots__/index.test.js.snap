// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 2. unknown plugin: 2. unknown plugin 1`] = `

a**=b

      ↓ ↓ ↓ ↓ ↓ ↓

var _a;

_a = a;
a = Math.pow(_a, b);


`;

exports[`unknown plugin 赋值: 赋值 1`] = `

a**=b

      ↓ ↓ ↓ ↓ ↓ ↓

var _a;

_a = a;
a = Math.pow(_a, b);


`;
